### DISCUSSION TITLE
Problem with Flow deck v2

### DISCUSSION THREAD

**Discussion by iulianoroberto** [bitcraze/discussions#802] [Q&A] [answered]
Hi, I'm having issues with Flow deck v2 after unplugging it and plugging it back into the Crazyflie.
Before I unplugged it everything was working fine.

This is log:
`SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:7d40b5411085 (2023.06) CLEAN
SYS: I am 0x20373435484350120045001A and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcFlow2 for deck 0
ZR2: Z-down sensor [FAIL]
PMW: Motion chip id: 0x:0x
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2023.02
EEPROM: I2C connection [FAIL].
SYS: configblock [FAIL]
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
Error while initializing the PMW3901 sensor
DECK_CORE: Deck 0 test [FAIL].
SYS: deck [FAIL]`

I think the female deck connector is unsoldered, in fact it's a bit tilted. 
Tips on how to fix? Thank you!

![IMG_5407](https://github.com/bitcraze/discussions/assets/94120694/2f92d64d-183c-4985-a95c-46c92ddf3742)

**Comment by GBdangerous** [comment]
I had a similar problem but even worse, the female deck just run off. And when i plug the flow deck back again and power pn, the red light on the front-right arm just blink in every 2 second. So I am looking for a solution too :(

**Comment by krichardsson** [comment, marked as answer]
Unplugging decks from the Crazyflie requires a bit of love and care, avoid to much tilt. The right connector was obviously bent a bit, but I can also see in the image that your left connector seems to have been damaged at some point. If you are lucky, you can just bend the right connector back, but there is a risk that some of the tabs in the connector are broken.  

There are [spare connectors in our store](https://store.bitcraze.io/collections/spare-parts-crazyflie-2-0/products/female-deck-connector) if you want to try to solder new ones.

**Reply by simondlevy** [reply]
![20230812_011418](https://github.com/bitcraze/discussions/assets/6766537/c3c4beed-a131-439b-a784-b60107ee8961)
![20230812_011449](https://github.com/bitcraze/discussions/assets/6766537/00695bd3-0d9d-4653-a59a-c969df720124)
As an improvement over the somewhat sloppy method I showed in the video I made earlier, here's a little deck (re)soldering rig that's easy to make from a [2mm-pitch proto board](https://www.amazon.com/dp/B097NLXVDF?psc=1&ref=ppx_yo2ov_dt_b_product_details) and [2mm-pitch pin headers](https://www.amazon.com/Uxcell-a16031600ux0732-Straight-Single-Breakaway/dp/B01GKJZ310/ref=sr_1_5?crid=3RMIDQNNGZRQC&keywords=2mm+pitch+pin+headers&qid=1691817877&s=electronics&sprefix=2mm+pitch+pin+header%2Celectronics%2C69&sr=1-5) soldered onto the proto board.   As shown in the photos here (using a custom deck I'm working on), this rig makes it easy to line up your deck with the [female deck connectors](https://store.bitcraze.io/products/female-deck-connector) for a good solder joint.
